client/src/components/LinkTo.js

```diff
@@ -5,6 +5,24 @@ import * as Models from "models"
 import PropTypes from "prop-types"
 import React, { Component } from "react"
 import { Link } from "react-router-dom"
+import { Tooltip } from "@blueprintjs/core"
+import AuthorizationGroupShow from "pages/admin/authorizationgroup/Show"
+import LocationShow from "pages/locations/Show"
+import OrganizationShow from "pages/organizations/Show"
+import PersonShow from "pages/people/Show"
+import PositionShow from "pages/positions/Show"
+import ReportShow from "pages/reports/Show"
+import TaskShow from "pages/tasks/Show"
+
+export const MODEL_TO_COMPONENT = {
+  [Models.AuthorizationGroup]: AuthorizationGroupShow,
+  [Models.Location]: LocationShow,
+  [Models.Organization]: OrganizationShow,
+  [Models.Person]: PersonShow,
+  [Models.Position]: PositionShow,
+  [Models.Report]: ReportShow,
+  [Models.Task]: TaskShow
+}
```

Comment:
Is it possible to use a specific preview component here (`TaskPreview` etc.), extract the specific part you want to see as part of the preview from the Show page into it, and include that extracted Preview component in the Show page?

Reply:
Yes, adds a little bit of repeating but we can work with it much easier. Query less data etc...